### PR TITLE
Active dur fix

### PR DESCRIPTION
**Pull request by tannermpeterson**

### CODE REVIEW

File: store/modules/stimulation/actions.js [hunk 1]
```diff
@@ -85,6 +78,13 @@ export default {
       color_assignments[color] = [starting_repeat_idx, ending_repeat_idx];
     });
 
+    // convert x_values to correct unit
+    console.log("@@@", state.x_axis_unit_name, TIME_CONVERSION_TO_MILLIS[state.x_axis_unit_name]);
```
Comment (luciipak) [inline store/modules/stimulation/actions.js:82]: console.logs

File: store/modules/stimulation/actions.js [hunk 2]
```diff
@@ -333,7 +333,9 @@ export default {
     }
     commit("reset_protocol_editor");
   },
-  handle_x_axis_unit({ commit, dispatch, state }, idx) {
+  handle_x_axis_unit({ commit, dispatch, state }, { idx, unit_name }) {
+    console.log("!!!", unit_name);
```
Comment (luciipak) [inline store/modules/stimulation/actions.js:337]: console.logs

File: components/stimulation/StimulationStudioDelayModal.vue [hunk 3]
```diff
@@ -162,16 +161,21 @@ export default {
 
       this.$emit("delay_close", button_label, delay_settings, stim_settings, frequency);
     },
-    check_validity(value) {
-      const number_regex = new RegExp("^([0]{1}.{1}[0-9]+|[1-9]{1}[0-9]*.{1}[0-9]+|[0-9]+|0)$");
+    check_validity(valueStr) {
```
Comment (luciipak) [inline components/stimulation/StimulationStudioDelayModal.vue:164]: snake case

File: components/stimulation/StimulationStudioWaveformSettingModal.vue [hunk 4]
```diff
@@ -458,76 +593,109 @@ export default {
...
-    check_charge_validity(value, label) {
-      if (value === "") {
-        this.err_msg[label] = this.invalid_err_msg.required;
-      } else if (!this.regex.charge.test(value)) {
-        this.err_msg[label] = this.invalid_err_msg.num_err;
-      } else if (this.stimulation_type.includes("C") && (-100 > value || 100 < value)) {
-        this.err_msg[label] = this.invalid_err_msg.max_current;
-      } else if (this.stimulation_type.includes("V") && (-1200 > value || 1200 < value)) {
-        this.err_msg[label] = this.invalid_err_msg.max_voltage;
+    check_charge_validity(valueStr, label) {
```
Comment (luciipak) [inline components/stimulation/StimulationStudioWaveformSettingModal.vue:663]: snake case

**Review by luciipak — APPROVED** [review]

File: components/stimulation/StimulationStudioWaveformSettingModal.vue (hunk 4 above)
Comment (tannermpeterson) [reply components/stimulation/StimulationStudioWaveformSettingModal.vue:663]: We're using snake case in this repo

**Pull request merged**